Add download glyph to EuiIcon

miukimiu opened the pull request:
### Summary

Closes #3361.

- This PR adds a download glyph to **EuiIcon**. 
- I'm using the same arrow from the **export** and **share** glyphs. So for consistency, I did not change its size.

### Dark and light themes

<img width="725" alt="download-icon-themes@2x" src="https://user-images.githubusercontent.com/2750668/80019907-24948a80-84d0-11ea-83d5-2136687f1e29.png">

### Design

<img width="646" alt="Screenshot 2020-04-22 at 19 06 55" src="https://user-images.githubusercontent.com/2750668/80019867-15add800-84d0-11ea-8cf2-c9421e94ecdf.png">

### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
- [ ] ~~Checked in **mobile**~~
- [ ] ~~Checked in **IE11** and **Firefox**~~
- [ ] ~~Props have proper **autodocs**~~
- [x] Added **documentation** examples
- [x] Added or updated **jest tests**
- [ ] ~~Checked for **breaking changes** and labeled appropriately~~
- [ ] ~~Checked for **accessibility** including keyboard-only and screenreader modes~~
- [x] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately

cchaos commented:
Are we worried that there might be conflict between the understanding of this icon and the current `importAction`?
<img width="246" alt="Screen Shot 2020-04-21 at 08 48 38 AM" src="https://user-images.githubusercontent.com/549577/79867838-f702f080-83ac-11ea-9da7-00bcd57e41b0.png">

They're very similar but mean very different things.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3364/

snide commented:
We discussed this today in our meeting. We decided it's OK to have them slightly different. There might be a followup to change the "export" icon to separate these concerns a little bit more.

snide commented:
We discussed this today in our meeting. We decided it's OK to have them slightly different. There might be a followup to change the "export" icon to separate these concerns a little bit more.

cchaos commented:
The one thing you'll want to tweak here is that the main line of the arrow cannot straddle 2 pixels. It has to align inside. This will make your icon not be fully center and 1px smaller than the whole artboard, but that is ok. The straddling will make the line very fuzzy for non-retina screens. Here's the example from the import icon

<img width="449" alt="Screen Shot 2020-04-22 at 09 55 28 AM" src="https://user-images.githubusercontent.com/549577/79990955-b031fb00-847f-11ea-82fe-646847f7665a.png">

Also, don't forget to add this to the Figma EUI icon set.

miukimiu commented:
@cchaos I've tweaked the icon. 

Now the download icon is not centered anymore but the main line of the arrow is not straddling. Based on other similar icons I also left a 1px space below the icon.

I also added the icon to the Figma icon set and updated the PR description with new screenshots.

<img width="901" alt="Screenshot 2020-04-22 at 19 21 37" src="https://user-images.githubusercontent.com/2750668/80018845-923fb700-84ce-11ea-81c7-41ddf23c1939.png">

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3364/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3364/

nicolapagni commented:
Hi,

since the icon should be merged into the master, using @elastic/eui 24.0.0 and using this type of icon I don't see that correctly.
<img width="146" alt="download-icon-error" src="https://user-images.githubusercontent.com/17281895/83635059-dc26bd00-a5a3-11ea-9e63-b22926e51457.png">

miukimiu commented:
Hi @nicolapagni, 

Thank you for reporting this error. Can you please open an [issue](https://github.com/elastic/eui/issues/new)  and specify with a code example how are you using the icon? 

Thanks!

nicolapagni commented:
@miukimiu done.

https://github.com/elastic/eui/issues/3543
